### PR TITLE
Submitted application overview page

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

https://trello.com/c/C3cuBfkw/587-nacro-application-overview

# Changes in this PR

## Screenshots of UI changes

Screenshot 2023-11-03 at 14 11 33

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/views/assess/applications/overview.njk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{% set pageTitle = applicationName + " - " + pageHeading  %}
+{% set mainClasses = "app-container govuk-body" %}
+
+{% block content %}
+
+  <h1 class="govuk-heading-l">
+    {{ application.person.name }}
+  </h1>
+
+  <p>This application was submitted on {{ formatDate(application.submittedAt, {format: 'medium'}) }}.</p>
```
Comment (Harriethw) [inline server/views/assess/applications/overview.njk:20]: v minor but I think the date should be in bold

**Review by Harriethw — APPROVED** [review]
all killer no filler 🔥 
would be interested just to get Ed's double tick on whether we understood his vision for the controller :)

**Review by edavey — APPROVED** [review]
Nice work! 

I've left a comment on the new "assess" path definition as I think you can make that simpler and more accurately represent the new custom (non-REST) 'overview' action.

File: server/views/assess/applications/overview.njk (hunk 1 above)
Comment (patrickjfl) [reply server/views/assess/applications/overview.njk:20]: oh yes good spot

File: server/controllers/assess/submittedApplicationsController.test.ts [hunk 2]
```diff
@@ -45,4 +45,23 @@ describe('submittedApplicationsController', () => {
...
+
+  describe('overview', () => {
+    it('renders the submitted application overview template', async () => {
+      submittedApplicationService.findApplication.mockResolvedValue(submittedApplication)
+      const person = submittedApplication.person as FullPerson
+
+      const requestHandler = submittedApplicationsController.overview()
+      await requestHandler(request, response, next)
+
+      expect(paths.submittedApplications.overview.show({ id: submittedApplication.id })).toEqual(
```
[the remark this answers is not in the document]
Comment (patrickjfl) [reply server/controllers/assess/submittedApplicationsController.test.ts:57]: that makes sense, will amend now

File: integration_tests/tests/assess/view_application_overview.cy.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  // ----------------------------------------------
+  it('displays submitted application overview', () => {
+    // And I am logged in as a NACRO assessor
+    cy.signIn()
+
+    // When I visit the application overview for that application
+    const page = SubmittedApplicationOverviewPage.visit(submittedApplication)
+
+    // Then I see the submitted application overview
+    page.shouldShowApplicationDetails(submittedApplication)
```
[the remark this answers is not in the document]
Comment (patrickjfl) [reply integration_tests/tests/assess/view_application_overview.cy.ts:39]: yes, agree

**Pull request merged**